tensorboard/webapp/widgets/dropdown/dropdown_test.ts

```diff
@@ -0,0 +1,148 @@
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [MatSelectModule],
+      declarations: [DropdownComponent, TestableComponent],
+    }).compileComponents();
+  });
+
+  function createTestableComponent(input: {
+    expId?: string;
+    configOptions?: DropdownOption[];
+  }): ComponentFixture<TestableComponent> {
+    const fixture = TestBed.createComponent(TestableComponent);
+    fixture.componentInstance.expId = input.expId || '';
+    fixture.componentInstance.configOptions = input.configOptions || [];
+    selectionChangeSpy = jasmine.createSpy();
+    fixture.componentInstance.onSelectionChange = selectionChangeSpy;
+    return fixture;
+  }
+
+  it('renders options with no aliases', () => {
+    let fixture = createTestableComponent({
+      expId: '1',
+      configOptions: [
+        {
+          value: '',
+          displayText: 'none',
+          disabled: true,
+        },
+        {
+          value: '1',
+          displayText: 'abc',
+          disabled: false,
+          displayAlias: '', // empty aliases will be ignored
+        },
+      ],
+    });
+    fixture.detectChanges();
+
+    const dropdown = fixture.debugElement.query(byCss.DROPDOWN);
+    expect(dropdown.componentInstance.value).toEqual('1');
+    expect(dropdown.componentInstance.options).toEqual([
+      jasmine.objectContaining({
+        value: '',
+        displayText: 'none',
+        disabled: true,
+      }),
+      jasmine.objectContaining({value: '1', displayText: 'abc'}),
+    ]);
+    // TODO(ytjing): Figure out how to test DOM content.
+  });
+
+  it('renders options with valid aliases', () => {
+    let fixture = createTestableComponent({
+      expId: '2',
+      configOptions: [
+        {
+          value: '2',
+          displayText: 'test experiment name',
+          disabled: false,
+          displayAlias: 'test alias',
+        },
+      ],
+    });
+    fixture.detectChanges();
+
+    const dropdown = fixture.debugElement.query(byCss.DROPDOWN);
+    expect(dropdown.componentInstance.value).toEqual('2');
+    expect(dropdown.componentInstance.options).toEqual([
+      jasmine.objectContaining({
+        value: '2',
+        displayText: 'test experiment name',
+        displayAlias: 'test alias',
+      }),
+    ]);
+    // TODO(ytjing): Figure out how to test DOM content.
+  });
+
+  it('changes selection', async () => {
+    let fixture = createTestableComponent({
+      expId: '1',
+      configOptions: [
+        {
+          value: '1',
+          displayText: 'abc',
+        },
+        {
+          value: '2',
+          displayText: 'efg',
+        },
+      ],
+    });
+    fixture.detectChanges();
+
+    // Triggers selection change event.
+    const matSelect = fixture.debugElement.query(byCss.MAT_SELECT);
+    matSelect.triggerEventHandler('selectionChange', {value: '2'});
+    fixture.detectChanges();
+
+    expect(selectionChangeSpy).toHaveBeenCalledOnceWith('2');
```

Comment:
I think this should be called with `{value: '2;}` instead of '2'. Please check again.

Reply:
After checking the call arg appeared to be `'2'`, similar to the example: https://cs.opensource.google/tensorflow/tensorboard/+/master:tensorboard/webapp/widgets/range_input/range_input_test.ts;l=228?q=range_input&ss=tensorflow%2Ftensorboard